packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -212,3 +216,168 @@ class _TweenAnimationBuilderState<T extends Object?>
+  RepeatingTweenAnimationBuilder({
+    super.key,
+    required this.tween,
+    required this.duration,
+    this.curve = Curves.linear,
+    this.reverse = false,
+    this.paused = false,
+    required this.builder,
+    this.child,
+  }) : assert(tween.begin != null, 'Tween.begin must not be null'),
+       assert(tween.end != null, 'Tween.end must not be null');
+
+  /// The tween to animate.
+  ///
+  /// The animation continuously cycles from [Tween.begin] to [Tween.end].
+  ///
+  /// ## Ownership
+  ///
+  /// The [RepeatingTweenAnimationBuilder] takes ownership of the provided [Tween]
+  /// and may mutate it. Once a [Tween] instance has been passed to
+  /// [RepeatingTweenAnimationBuilder] its properties should not be accessed or
+  /// changed to avoid interference with the [RepeatingTweenAnimationBuilder].
+  ///
+  /// It is good practice to never store a [Tween] provided to a
+  /// [RepeatingTweenAnimationBuilder] in an instance variable to avoid accidental
+  /// modifications of the [Tween].
+  final Tween<T> tween;
+
+  /// The duration of the animation.
+  final Duration duration;
+
+  /// The curve to apply when animating the value.
+  ///
+  /// Defaults to [Curves.linear].
+  final Curve curve;
+
+  /// Called every time the animation value changes.
+  ///
+  /// The builder receives the animation object and the optional child.
+  /// The current value can be accessed via animation.value.
+  final Widget Function(BuildContext context, Animation<T> animation, Widget? child) builder;
+
+  /// The child widget to pass to the builder.
+  ///
+  /// If a builder callback's return value contains a subtree that does not
+  /// depend on the animation, it's more efficient to build that subtree once
+  /// instead of rebuilding it on every animation tick.
+  final Widget? child;
+
+  /// Whether the animation should reverse direction when it reaches the end.
+  ///
+  /// When true, the animation reverses direction instead of jumping back to
+  /// the beginning. Defaults to false.
+  final bool reverse;
+
+  /// Whether the animation is currently paused.
+  ///
+  /// When true, the animation stops at its current value. When changed to
+  /// false, the animation resumes from that value. Defaults to false.
+  final bool paused;
+
+  @override
+  State<RepeatingTweenAnimationBuilder<T>> createState() {
+    return _RepeatingTweenAnimationBuilderState<T>();
+  }
+}
+
+class _RepeatingTweenAnimationBuilderState<T extends Object>
+    extends State<RepeatingTweenAnimationBuilder<T>>
+    with SingleTickerProviderStateMixin {
+  late final AnimationController _controller;
+  late Animation<T> _animation;
+  CurvedAnimation? _curvedAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(duration: widget.duration, vsync: this);
+    _updateAnimation();
+
+    // Start the animation if not paused and begin != end
```

Comment:
Please add a period to the end of all comments :)

```suggestion
    // Start the animation if not paused and begin != end.
```